### PR TITLE
Nip5 shop

### LINKED ISSUE #37: NIP-05 Shop
Allow creation of NIP-05 links directly from profile page (free/paid services to be connected)

### PR DESCRIPTION
**Pull request by v0l**
Adds NIP-05 integration with https://nostrplebs.com

Closes #37

### CODE REVIEW

File: src/element/Nip5Service.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            setShowInvoice(true);
+        }
+    }
+
+    async function updateProfile(handle: string, domain: string) {
+        let newProfile = {
+            ...user,
+            nip05: `${handle}@${domain}`
+        };
+        debugger;
```
Comment (verbiricha) [inline src/element/Nip5Service.tsx:149]: 🔥

File: src/element/Nip5Service.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    ServiceProvider,
+    ServiceConfig,
+    ServiceError,
+    HandleAvailability,
+    ServiceErrorCode,
+    HandleRegisterResponse,
+    CheckRegisterResponse
+} from "../nip05/ServiceProvider";
+import AsyncButton from "./AsyncButton";
+// @ts-ignore
```
Comment (verbiricha) [inline src/element/Nip5Service.tsx:14]: We can revisit this once our components are in TS.

**Review by verbiricha — APPROVED** [review]
Apart from the `debugger` statement LGTM :shipit: Are we going to diplay these NIP-05 addresses differently from others in the future too?

**Comment by v0l** [comment]
> Apart from the `debugger` statement LGTM :shipit: Are we going to diplay these NIP-05 addresses differently from others in the future too?

The nostr plebs ones? probably. For snort we will have some gold badge maybe idk

**Pull request merged**